File: Codeforces/contest/1283/D.cpp
#include <bits/stdc++.h>

using namespace std;

#define N int(2e5) + 10
int a[N];
int b[N];
map<int, int> s;
queue<int> q;
long long ans = 0;
int cnt =0;
int main() {
	int n, m;
	cin >> n >> m;
	for (int i = 0; i < n; i++) {
		cin >> a[i];
		q.push(a[i]);
		s[a[i]] = 0;
	}
	while (!q.empty()) {
		int top = q.front();
		q.pop();
		if(s.count(top+1) == 0){
			ans += s[top] +1;
			q.push(top+1);
			s[top+1] = s[top] +1;
			b[cnt] = top +1;
			cnt += 1;
		}
		if ( cnt == m){
			break;
		}
		if(s.count(top-1) == 0){
			ans += s[top] +1;
			q.push(top-1);
			s[top-1] = s[top] +1;
			b[cnt] = top -1;
			cnt += 1;
		}
		if ( cnt == m){
			break;
		}
	}
	cout<<ans<<endl;
	for(int i=0;i<m;i++){
		cout<<b[i] <<" ";
	}
	cout<<endl;

	return 0;
}

File: Codeforces/contest/1283/D.py
import queue
q = queue.Queue()
occupied = {}
n, m = map(int, input().split())
a = list(map(int, input().split()))
for i in a:
    occupied[i] = 0
    q.put(i)

print
sum_consume = 0
cnt = 0
output = []
while not q.empty():
    top = q.get()
    if top-1 not in occupied:
        sum_consume += occupied[top]+1
        occupied[top-1] = occupied[top] +1 
        cnt += 1
        output.append(top-1)
        q.put(top-1)
    if cnt == m:
        break

    if top+1 not in occupied:
        sum_consume += occupied[top]+1
        occupied[top+1] = occupied[top] +1 
        cnt += 1
        output.append(top+1)
        q.put(top+1)
    if cnt == m:
        break
print(sum_consume)

print(" ".join(str(i) for i in output))
